static/js/recent_topics_ui.js

```diff
@@ -344,31 +398,37 @@ export function update_topics_of_deleted_message_ids(message_ids) {
 export function filters_should_hide_topic(topic_data) {
     const msg = message_store.get(topic_data.last_msg_id);
     const sub = sub_store.get(msg.stream_id);
-
-    if (sub === undefined || !sub.subscribed) {
+    if ((sub === undefined || !sub.subscribed) && topic_data.type !== "private") {
         // Never try to process deactivated & unsubscribed stream msgs.
         return true;
     }
```

Comment:
This needs a condition to check the message is a stream message and remove the ` topic_data.type !== "private"` condition from here since we are still processing `unread` private messages.

Reply:
I have changed it to `topic_data.type === "stream"`. Unread PM's are still processed with this.